Fix visibility of Tab's close button when onClosed is null

diluculo opened the pull request:
Added missing `isClosable` property to `Tab` widget to match WinUI 3's [TabViewItem.IsClosable](https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.tabviewitem.isclosable?view=winui-2.8) implementation. 

The property defaults to `true`. When `isClosable` is `false`:
- Close button is not shown
- Keyboard shortcuts (Ctrl+W, Ctrl+F4) are disabled

<!-- Mark all that applies -->

- [x] I have updated `CHANGELOG.md` with my changes <!-- REQUIRED --> 
- [x] I have run "dart format ." on the project <!-- REQUIRED --> 
- [x] I have added/updated relevant documentation

bdlukaa commented:
We use a different pattern in Flutter. If the `onClose` callback is null, it is automatically marked as disabled, therefore there is no need for another property.

diluculo commented:
Thank you for the feedback about `onClosed`.

I was actually trying to find a way to hide the close button in certain tabs. Looking at the current code, I noticed that the close button is shown even when `onClosed` is null.

I simplified the solution and just added a null check for `onClosed`.

![image](https://github.com/user-attachments/assets/843f3946-6546-44d9-8b38-219cb4abe2a3)

In this image, the first tab has `onClosed = null` and with the added null check, its close button is now properly hidden while other tabs show their close buttons as expected.

bdlukaa reviewed (commented):
Can you add it to the changelog?

diluculo commented:
@bdlukaa, Done! I added the fix to the changelog and also squashed the commits into a single one.
